feature-libs/quote/components/actions/by-role/quote-actions-by-role.component.ts

```diff
@@ -0,0 +1,229 @@
+  protected globalMessageService = inject(GlobalMessageService);
+  protected config = inject(QuoteUIConfig);
+  protected activeCartFacade = inject(ActiveCartFacade);
+
+  quoteDetails$: Observable<Quote> = this.quoteFacade.getQuoteDetails();
+  cartDetails$: Observable<Cart> = this.activeCartFacade.getActive();
+
+  @ViewChild('element') element: ElementRef;
+  QuoteActionType = QuoteActionType;
+  protected subscription = new Subscription();
+
+  ngOnInit(): void {
+    //submit button present and threshold not reached: Display message
+    this.quoteDetails$.pipe(take(1)).subscribe((quote) => {
+      const mustDisableAction = quote.allowedActions.find((action) =>
+        this.mustDisableAction(action.type, quote)
+      );
+      if (mustDisableAction) {
+        this.globalMessageService.add(
+          {
+            key: 'quote.commons.minRequestInitiationNote',
+            params: {
+              minValue: quote.threshold,
+            },
+          },
+          GlobalMessageType.MSG_TYPE_WARNING
+        );
+      }
+    });
+  }
+
+  mustDisableAction(type: string, quote: Quote): boolean {
+    return type === QuoteActionType.SUBMIT && !this.isThresholdReached(quote);
+  }
+
+  protected isThresholdReached(quote: Quote): boolean {
+    const requestThreshold = quote.threshold || 0;
+    return (quote.totalPrice.value || 0) >= requestThreshold;
+  }
+
+  onClick(action: QuoteActionType, quote: Quote, cart: Cart) {
+    const cartIsEmpty = (cart.entries?.length ?? 0) === 0;
+    if (!this.isConfirmationDialogRequired(action, quote.state, cartIsEmpty)) {
+      this.performAction(action, quote);
+      return;
+    }
+    const context = this.prepareConfirmationContext(action, quote);
+    this.launchConfirmationDialog(context);
+    this.handleConfirmationDialogClose(action, context);
+  }
+
+  protected performAction(action: QuoteActionType, quote: Quote) {
+    if (action === QuoteActionType.REQUOTE) {
+      this.requote(quote.code);
+    } else {
+      this.quoteFacade.performQuoteAction(quote, action);
+    }
+  }
+
+  protected launchConfirmationDialog(context: ConfirmationContext) {
+    this.launchDialogService
+      .openDialog(
+        LAUNCH_CALLER.ACTION_CONFIRMATION,
+        this.element,
+        this.viewContainerRef,
+        { confirmationContext: context }
+      )
+      ?.pipe(take(1))
+      .subscribe();
+  }
+
+  protected handleConfirmationDialogClose(
+    action: QuoteActionType,
+    context: ConfirmationContext
+  ) {
+    this.subscription.add(
+      this.launchDialogService.dialogClose
+        .pipe(
+          filter((reason) => reason === 'yes'),
+          tap(() => this.performAction(action, context.quote)),
+          filter(() => !!context.successMessage),
+          tap(() =>
+            this.globalMessageService.add(
+              { key: context.successMessage },
+              this.getMessageType(action)
+            )
+          )
+        )
+        .subscribe()
```

Comment:
@Uli-Tiger Suppose an action is performed x times, the launch dialogue always aborted. Then we will subscribe x times. On time x+1, when the user finally choses to go ahead, the subscription fires x+1 times. Subscription should only fire once

Reply:
you are right -fixed it.